x/subspaces/keeper/msg_server.go

```diff
@@ -0,0 +1,316 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/desmos-labs/desmos/v2/x/subspaces/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the stored MsgServer interface
+// for the provided keeper
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+// CreateSubspace defines a rpc method for MsgCreateSubspace
+func (k msgServer) CreateSubspace(goCtx context.Context, msg *types.MsgCreateSubspace) (*types.MsgCreateSubspaceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Get the next subspace ID
+	subspaceID, err := k.GetSubspaceID(ctx)
```

Comment:
Are you sure of this approach for the ID? I remember we first used the same in our first `posts` implementation and then we switched to a SHA-256 id to avoid duplication if I'm not wrong.

Reply:
Yes, I think duplication will not be a problem. The subspaces are thought to be created by people who want to create a DApp. The idea is that you want to build something on Desmos and thus you create the subspace. Later you will have all your clients that rely on that subspace that are marketed as "official" for such subspace. Even if someone creates a copy subspace, they would have to have their clients relying on the new subspace instead of the official one, which will be quite a lot of work. 

Also, the `treasury` address is used to allow the creators to verify their subspace with external applications (eg. their website) so that other people cannot create copies of it that easily. It's the same principle of app links but applied to subspaces: the more links you add, the safer and easier for users is to verify the subspace is what it is claiming to be.